src/cpp/participant/ParticipantFactory.cpp

```diff
@@ -36,54 +36,79 @@ namespace ddsrouter {
 // TODO: Add logs
 
 std::shared_ptr<IParticipant> ParticipantFactory::create_participant(
-        ParticipantConfiguration participant_configuration,
+        std::shared_ptr<configuration::ParticipantConfiguration> participant_configuration,
         std::shared_ptr<PayloadPool> payload_pool,
         std::shared_ptr<DiscoveryDatabase> discovery_database)
 {
+    // This switch needs to create different kind of elements (ParticipantConfigurations) inside some of the cases
+    // and C++ does not like it. Thus, they must be inside blocks {}, but C++ does not like it either.
+    // Thus, there must be a final tsnh line.
```

Comment:
Some breaks are missing, could it be it?

Reply:
could be. The thing is that switch case behaves weird when using blocks inside, and does not understand correctly a break or return.
In a normal switch, a return acts as well as a break.

Reply:
I see. Well, maybe add all the breaks even if in principle they are not required, just in case.